Rename cluster to addCluster

theotimeee opened the pull request:
Rename `cluster` to `addCluster.` 
Fix indentation in calculateDMM and cluster files following Bioconductor best practices (indentation must be a multiple of 4 spaces).
Move deprecated functions from the calculateDMM file to deprecate file.

Commit:
rename cluster to addCluster

fix identation

Commit:
documentation

Commit:
up

Commit:
up

Commit:
up

Commit:
merge master

Merge branch 'master' of https://github.com/microbiome/mia into rename_cluster

# Conflicts:
#	R/deprecate.R
#	man/deprecate.Rd

Commit:
up

Merge branch 'master' of https://github.com/microbiome/mia into rename_cluster

# Conflicts:
#	R/cluster.R
#	tests/testthat/test-9cluster.R

Commit:
version update + news update

Commit:
up

TuomasBorman commented on R/cluster.R:
```diff
@@ -112,3 +113,49 @@ setMethod("cluster", signature = c(x = "SummarizedExperiment"),
     }
 )
+
+.get_altExp <- function(x, ...) {
+    altexppos <- which(...names() == "altexp")
+    if (length(altexppos) == 0) {
+        altexp <- NULL
+    } else {
+        altexp <- ...elt(altexppos)
+    }
+    se <- .check_and_get_altExp(x, altexp)
+    list(se = se, altexp = altexp)
+}
+
+.check_margin <- function(MARGIN) {
+    if (.is_non_empty_string(MARGIN)) {
+        MARGIN <- tolower(MARGIN)
+    }
+    if (length(MARGIN) != 1L 
+        || !(MARGIN %in% c(1, 2, "features", "samples", "columns", 
+                            "col", "row", "rows", "cols"))) {
+        stop("'MARGIN' must equal to either 1, 2, 'features', 'samples', 'columns', 'col', 'row', 'rows', or 'cols'.",
+            call. = FALSE)
+    }
+    MARGIN <- ifelse(MARGIN %in% c("samples", "columns", "col", 2, "cols"), 
+                    2, 1)
+    MARGIN
+}
+
+.check_name <- function(x, name) {
+    if (name %in% names(metadata(x))) {
+        stop("The 'name' must not exist in the metadata of the object.", call. = FALSE)
+    }
+}
+
+.check_data_name <- function(x, clust.col, MARGIN) {
+    if (MARGIN == 1) {
+        if (clust.col %in% names(rowData(x))) {
+            stop("The 'clust.col' parameter must not exist in the names of the rowData of the object.", 
+                call. = FALSE)
```
Sorry, I made a commit just couple hours ago where these functions were removed. Sync with master.

Commit:
up

Merge branch 'master' of https://github.com/microbiome/mia into rename_cluster

# Conflicts:
#	NEWS

Commit:
correct merge master into rename_cluster

TuomasBorman commented on R/deprecate.R:
```diff
@@ -1,12 +1,52 @@
 #' These functions will be deprecated. Please use other functions instead.
 #' 
+#' @param x A
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param clust.col A single character value indicating the name of the 
+#'   \code{rowData} (or \code{colData}) where the data will be stored.
+#'   
+#' @param exprs_values a single \code{character} value for specifying which
+#'   assay to use for calculation.
+#'   (Please use \code{assay.type} instead.)
+#'   
+#' @param transposed Logical scalar, is x transposed with cells in rows?
+#' 
+#' @param type the type of measure used for the goodness of fit. One of
```
This manual page should collect all deprecated functions. Some of them might have same parameter name but different meaning. So I'm wondering what happens then. --> I'm wondering if it is better to not list parameters for any of these functions since user's should not use them anyways (even though BIocCheck would tell otherwise)

TuomasBorman commented on R/deprecate.R:
```diff
@@ -1,12 +1,52 @@
+                    name = "clusters", clust.col = "clusters", ...){
+                .Deprecated(msg = paste0("'cluster' is deprecated. ",
+                                        "Use 'addCluster' instead."))
+                addCluster(x)
```
This does not pass parameters. addCluster gets only x parameter. Not others. Moreover, you can skip listing parameters. Use ... and then pass them to addCluster with ...


Only x is needed to list since the method is chosen based on class of x

TuomasBorman replied:
The idea is to keep these functions a hollow as possible. Just that they work but do not do anything but call the new function

Commit:
up

Commit:
up

Commit:
up

Commit:
up

Commit:
keep deprecated calculateDMM functions in calculateDMM file

Commit:
up

Commit:
correct indentation in calcucateDMM file

Commit:
correct indentation calculateDMM file

TuomasBorman reviewed (approved):
Looks good! @antagomir

antagomir commented:
There are unresolved comments, and one conflict. If those are resolved then this seems good to me at least?

Commit:
resolve conflict NEWS file

Merge branch 'devel' of https://github.com/microbiome/mia into rename_cluster

# Conflicts:
#	NEWS

Commit:
up

Commit:
add missing topic
